### DISCUSSION TITLE
Adding and editing sketches in form of svgs

### DISCUSSION THREAD

**Discussion by ModProg** [GitJournal/GitJournal#713] [Feature Requests]
## What you'd like to happen:
1. Adding a 3rd option to the attach button `Sketch` which opens a simple Sketching plane where you can draw with touch (or S-Pen etc.) and then attach it as an SVG.
2. When a Note contains a compatible SVG (This could be marked by the `[Name]` some meta information in the svg or any SVG which only consists of supported features. Make them editable, not sure yet how:
 - Onpress in rendered view
 - Toolbar button in edit mode to either:
  - Open Menu to select sketch from
  - Select the `[...](... .svg)` line to edit
3. Add option to render sketches in edit mode as well (onpress could then open them as well)

## Alternatives you've considered:
Sadly there exists nothing like this (at leas on android)

**PS: I will work on this feature myself, but like to have the discussion here**

**Comment by ModProg** [comment]
My plan for this is:
I will develop an companion  App for creating and editing simple Sketches. This will probably be implemented in Godot, as I should have all the needed Features there, and can build on https://github.com/mbrlabs/Lorien (MIT)

Features required in GitJournal:
- [ ] Open File in external Application (somthing along the lines of #340)
- [x] SVG Rendering (alread implemented #246)

Requirements collection for the Sketching:
- SVG conform
- Additional "hidden" Information comparable to Inkscape (not compatible does not make sense) to support non standard Features (e.g. variable width lines)

**Sadly SVG conformity will be mostly one sided, ie. you can view the sketches but not edit them outside the sketching App**

Repositories:
- Rust Library for parsing: https://github.com/ModProg/SVGNotesLib
- Godot Project (Currently in experimenting mode :D) https://github.com/ModProg/svg-note

Planned Features:
- [ ] Open File from another Application (Not sure how that works exactly on Mobile, needs to be investigated)
- [ ] Pen Pressure (Godot has iOS support, Android PR: godotengine/godot#49609
- [ ] Drawing
  - [x] Pen
    - [ ] Support Pressure in SVG
  - [ ] Lines
  - [ ] Simple Geometric Shapes (Square/Circle/Triangle)
  - [ ] Textmarker
- [ ] Eraser
  - [ ] Line Erasing
  - [ ] "Normal" Eraser
- [ ] Color Picker
- [ ] Spacer (like in Xournal++)
- [ ] Selection
  - [ ] Lasso
  - [ ] Box
  - [ ] "Drag Over" (Just drag over the lines you want to select, easier but less precise than Laso)
  - [ ] Actions 
    - [ ] Move
    - [ ] Delete
    - [ ] Change Color
    - [ ] Change Thickness

There are a lot more features that could be interresting. 
This is more meant as a sketching Tool not as a note taking Tool.

**Comment by vHanda** [comment]
This sounds awesome. Please let me know if I can help.
